java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistEnumDeclaration.java

```diff
@@ -233,24 +234,35 @@ public boolean hasInternalType(String name) {
         return this.internalTypes().stream().anyMatch(f -> f.getName().endsWith(name));
     }
 
+    @Deprecated
```

Comment:
Why is this deprecated? Should we add a comment for users?

Reply:
Its siblings in JavassistClassDeclaration and JavassistInterfaceDeclaration were already deprecated (back in 2016). I intended to tag this method as deprecated when I added it, but forgot. The reason for deprecation, as far as I see it, is that this family of methods should be refactored a bit to move the algorithmic (common) part into SymbolSolver and leave the classes to provide the details (like superclass FQN). This would ensure the same feature set across all types of source material. I intend to work on this refactor as part of issue #326 as well, but there are some other things that need to happen first.

Note: I’m not 100% sure if this was the original intent of the deprecated annotation on the siblings.

Reply:
Here's my random deprecation tip: when adding @Deprecated anywhere, add @deprecated to the Javadoc with a clear explanation why it is deprecated and what to use instead :-)

Reply:
Fair point, marking this as deprecated isn't correct in this case, as there is no replacement. Will fix!